listenbrainz/mbid_mapping/mapping/dump.py

```diff
@@ -0,0 +1,219 @@
+    try:
+        dump_location = create_mapping_dump(location, dump_time)
+    except IOError as e:
+        current_app.logger.critical(
+            'IOError while creating canonical metadata dump: %s', str(e), exc_info=True)
+        current_app.logger.info('Removing created files and giving up...')
+        shutil.rmtree(location)
+        return
+    except Exception as e:
+        current_app.logger.critical(
+            'Unable to create canonical metadata dump due to error %s', str(e), exc_info=True)
+        current_app.logger.info('Removing created files and giving up...')
+        shutil.rmtree(location)
+        return
+
+    current_app.logger.info(
+        'MusicBrainz canonical metadata PostgreSQL data dump created at %s!', location)
+    return dump_location
+
+
+def _create_dump(location: str, db_engine: Optional[sqlalchemy.engine.Engine], tables: Optional[dict],
+                dump_time: datetime):
+    """ Creates a dump of the provided tables at the location passed
+
+        Arguments:
+            location: the path where the dump should be created
+            db_engine: an sqlalchemy Engine instance for making a connection
+            tables: a dict containing the names of the tables to be dumped as keys and the columns
+                    to be dumped as values
+            dump_time: the time at which the dump process was started
+
+        Returns:
+            the path to the archive file created
+    """
+
+    archive_name = 'metabrainz-metadata-dump-{time}'.format(
+        time=dump_time.strftime('%Y%m%d-%H%M%S')
+    )
+    archive_path = os.path.join(location, '{archive_name}.tar.zst'.format(
+        archive_name=archive_name,
+    ))
+
+    with open(archive_path, 'w') as archive:
+
+        zstd_command = ["zstd", "--compress", "-10"]
+        zstd = subprocess.Popen(zstd_command, stdin=subprocess.PIPE, stdout=archive)
+
+        with tarfile.open(fileobj=zstd.stdin, mode='w|') as tar:
+
+            temp_dir = tempfile.mkdtemp()
+
+            try:
+                timestamp_path = os.path.join(temp_dir, "TIMESTAMP")
+                with open(timestamp_path, "w") as f:
+                    f.write(dump_time.isoformat(" "))
+                tar.add(timestamp_path,
+                        arcname=os.path.join(archive_name, "TIMESTAMP"))
+                tar.add(DUMP_LICENSE_FILE_PATH,
+                        arcname=os.path.join(archive_name, "COPYING"))
+            except IOError as e:
+                current_app.logger.error(
+                    'IOError while adding dump metadata: %s', str(e), exc_info=True)
+                raise
+            except Exception as e:
+                current_app.logger.error(
+                    'Exception while adding dump metadata: %s', str(e), exc_info=True)
+                raise
+
+            archive_tables_dir = os.path.join(temp_dir, 'metabrainz')
+            create_path(archive_tables_dir)
+
+            with db_engine.connect() as connection:
```

Comment:
You can use `db.engine.raw_connection()` to get a psycopg2 connection directly. Also, can probably also remove the explicit transaction stuff.

Reply:
Ah ok, I checked why this was added to main LB dumps. It is to ensure that all tables as visible at a given point are dumped. I guess that makes for these dumps as well.

Reply:
So you mean that it's best to keep the explicit transaction? In this case should we use raw_connection or stay with connect()?

Reply:
we discussed in IRC, given that having a transaction is useful we wil just leave this as-is